format/mbr/mbr.go

```diff
@@ -0,0 +1,66 @@
+package mbr
+
+import (
+	"fmt"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.MBR,
+		Description: "Master Boot Record",
+		DecodeFn:    mbrDecode,
+	})
+}
+
+func decodePartitionTableEntry(d *decode.D) {
+	d.FieldU8("boot_indicator", scalar.UToDescription{
+		0x80: "active",
+		0x00: "inactive",
+	})
+	d.FieldStrScalarFn("starting_chs_vals", decodeCHSBytes)
+	d.FieldU8("partition_type", partitionTypes)
+	d.FieldStrScalarFn("ending_chs_vals", decodeCHSBytes)
+	d.FieldStrScalarFn("starting_sector", decodeCHSBytes)
+	d.U8() // extra byte
+	d.FieldScalarU32("partition_size")
+}
+
+// Because this is a fixed-sized table, I am opting to use a
+// FieldStruct instead of a FieldArray
+func decodePartitionTable(d *decode.D) {
+	d.FieldStruct("entry_1", decodePartitionTableEntry)
+	d.FieldStruct("entry_2", decodePartitionTableEntry)
+	d.FieldStruct("entry_3", decodePartitionTableEntry)
+	d.FieldStruct("entry_4", decodePartitionTableEntry)
+}
+
+// Source: https://thestarman.pcministry.com/asm/mbr/PartTables.htm#Decoding
+func decodeCHSBytes(d *decode.D) scalar.S {
+	head, _ := d.Bits(8)
+	sectorHighBits, err := d.Bits(2)
+	if err != nil {
+		d.IOPanic(err, "chs")
+	}
```

Comment:
You can simplify this to just `d.U2()` or `d.U(2)` and it will do the error check and panic. I should probably rename´Bits` to `TryBits`, the convention usually is `Try*` functions returns error other just panic and there is usually pairs of functions.